### PR TITLE
Rans3pf

### PR DESCRIPTION
**Pull request by JohanMabille**
# Mandatory Checklist

Please ensure that the following criteria are met:

- [ ] Title of pull request describes the changes/features
- [ ] Request at least 2 reviewers
- [ ] If new files are being added, the files are no larger than 100kB. Post the file sizes.
- [ ] Code coverage did not decrease. If this is a bug fix, a test should cover that bug fix. If a new feature is added, a test should be made to cover that feature.
- [ ] New features have appropriate documentation strings (readable by sphinx)
- [ ] Contributor has read and agreed with CONTRIBUTING.md and has added themselves to CONTRIBUTORS.md 

As a general rule of thumb, try to follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

# Description

### CODE REVIEW

**Comment by cekees** [comment]
Maybe try compiling with PROTEUS_OPT="-w -g0 -O0 -DNDEBUG" ?

**Comment by JohanMabille** [comment]
I pushed to try, but are these flags acceptable in production? I mean, they would lead to a significative performance hit, no?

**Comment by JohanMabille** [comment]
The build is now passing, I need to fix the python file.

**Comment by cekees** [comment]
PROTEUS_OPT="-w -g0 -O0 -DNDEBUG" would just be for travis. My thinking is that the main objective of the travis build is to run the tests quickly. If errors arise then a developer can rebuild and test with PROTEUS_OPT="-w -g -O0 -UNDEBUG" and so on. I posted some comparisons of compile times on the proteus slack channel.

File: setup.py [hunk 1]
```diff
@@ -202,11 +202,13 @@ def __str__(self):
         libraries=['m',PROTEUS_LAPACK_LIB,
                    PROTEUS_BLAS_LIB]
     ),
-    Extension("mprans.cRANS3PF",['proteus/mprans/cRANS3PF.pyx'],
-              depends=['proteus/mprans/RANS3PF.h','proteus/mprans/RANS3PF2D.h', 'proteus/ModelFactory.h', 'proteus/CompKernel.h'],
-              language='c++',
-              extra_compile_args=PROTEUS_OPT+["-std=c++14","-mavx"],
-              include_dirs=get_xtensor_include()),
+    Extension(
+        'mprans.RANS3PF',
```
Comment (cekees) [inline setup.py:206]: mprans.RANS3PF ==> mprans.cRANS3PF

Comment (JohanMabille) [reply setup.py]: I'm gonna make a global review of these names, it looks like I missed some naming inconsistency between the setup.py, the XXX.py files and the XXX.cpp iles.

**Comment by cekees** [comment]
I think if you split the 2D and 3D bindings in to separate files, then it will fix our build problems. I'm just talking about RANS3PF/RANS3PF2D, RANS2P/RANS2P2D, and RANS3PSed/RANS3PSed2D:

https://pybind11.readthedocs.io/en/stable/faq.html#how-can-i-reduce-the-build-time

**Comment by JohanMabille** [comment]
I'll try that in a dedicated commit once the tests pass.

File: proteus/MeshAdaptPUMI/Adapter.cpp [hunk 2]
```diff
@@ -0,0 +1,14820 @@
+/* Generated by Cython 0.29.15 */
```
Comment (cekees) [inline proteus/MeshAdaptPUMI/Adapter.cpp:1]: @zhang-alvin we should delete this, right? This is just the cython generated file, not a pybind11 interface.

**Comment by JohanMabille** [comment]
Indeed, this was added by error with a greedy `git add proteus/`.

**Pull request merged**